alacritty/src/window_context.rs

```diff
@@ -470,30 +481,12 @@ impl WindowContext {
             self.mouse.hint_highlight_dirty = false;
         }
 
-        // Skip rendering until we get a new frame.
-        if !self.display.window.has_frame.load(Ordering::Relaxed) {
-            return;
-        }
-
-        if self.dirty && !self.occluded {
-            // Force the display to process any pending display update.
-            self.display.process_renderer_update();
-
-            self.dirty = false;
-
-            // Request immediate re-draw if visual bell animation is not finished yet.
-            if !self.display.visual_bell.completed() {
-                self.display.window.request_redraw();
-            }
-
-            // Redraw the window.
-            self.display.draw(
-                terminal,
-                scheduler,
-                &self.message_buffer,
-                &self.config,
-                &self.search_state,
-            );
+        // Request a redraw in case we need to draw a frame and we haven't asked for it yet..
```

Comment:
```suggestion
        // Request a redraw.
```

The second paragraph here already goes into more detail on this anyway.